nvdaHelper/remote/ia2LiveRegions.cpp

```diff
@@ -246,20 +247,29 @@ void CALLBACK winEventProcHook(HWINEVENTHOOK hookID, DWORD eventID, HWND hwnd, l
 	if(AccessibleObjectFromEvent(hwnd,objectID,childID,&pacc,&varChild)!=S_OK) {
 		return;
 	}
-	if (!useIAccessibleForLiveRegionUpdate(pacc, varChild)) {
+	//Retrieve an IAccessible2 via IServiceProvider if it exists.
+	pacc->QueryInterface(IID_IServiceProvider,(void**)(&pserv));
+	if(!pserv) {
 		VariantClear(&varChild);
 		pacc->Release();
 		return;
 	}
-	VariantClear(&varChild);
-	//Retreave an IAccessible2 via IServiceProvider if it exists.
-	pacc->QueryInterface(IID_IServiceProvider,(void**)(&pserv));
-	pacc->Release();
-	if(!pserv) return; 
 	pserv->QueryService(IID_IAccessible,IID_IAccessible2,(void**)(&pacc2));
 	pserv->Release();
-	if(!pacc2) return;
-	//Retreave the IAccessible2 attributes, and if the object is not a live region then ignore the event.
+	if(!pacc2) {
+		pacc->Release();

```

Comment:
Do you mean to check !pacc2  then release pacc?

Reply:
Yes, error paths undo everything that has been done up to that point. In this case, we have a live pacc and varChild so both are destroyed before returning.

Needless to say, the code would be more readable and less error-prone with holders / smart-pointers as compilers are really good at tracking variable lifetime ;)